src/shared/GitHub/GitHubOAuth2Client.cs

```diff
@@ -0,0 +1,33 @@
+using System;
+using System.Net.Http;
+using Microsoft.Git.CredentialManager.Authentication.OAuth;
+
+namespace GitHub
+{
+    public class GitHubOAuth2Client : OAuth2Client
+    {
+        private static readonly string ClientId = "0120e057bd645470c1ed";
+        private static readonly string ClientSecret = "18867509d956965542b521a529a79bb883344c90";
```

Comment:
secrets ??    not sure what the scope of these secrets are, but should we have a "TODO" to revisit and/or get them from a key vault of some kind?  again, not sure how these are used, so just asking.

Reply:
These are only really "secrets" if this OAuth application was used from a web server (confidential client) where it would have privileged access. For native apps (public client) this cannot be 'secret' since it will eventually need to exist in the application binary.

If you take a look inside something like the GitHub VS Extension, GitHub Desktop, or CLI, you'll find the ID and Secret baked into the assembly.